arches/app/datatypes/datatypes.py

```diff
@@ -2413,13 +2408,18 @@ def get_display_value(self, tile, node, **kwargs):
             except:
                 logger.info(f'Resource with id "{resourceid}" not in the system.')
         return ", ".join(items)
+    
+    def get_relationship_display_value(self, relationship_valueid):
+        concept_datatype_instance = self.datatype_factory.get_instance('concept')
+        concept_preflabel = concept_datatype_instance.get_pref_label(relationship_valueid)
```

Comment:
This will always return the preflabel in `en-US` because that's the default. You'll need to get the apps active language using something like this:

```
from django.utils.translation import get_language
lang = get_language()
```

All the datatype is doing in `get_pref_label` is calling a concept method. You could just call that same method and then you don't need the datatype:
```
from arches.app.models.concept import get_preflabel_from_valueid
get_preflabel_from_valueid(valid, lang)['value']
```

Reply:
Good catch on the language.
> All the datatype is doing in get_pref_label is calling a concept method. You could just call that same method and then you don't need the datatype

Yea I was reflecting on that for a little while. To be honest it still feels ambiguous to me whether the concept model's method should be called, or whether the concept datatype's method (which I recognize calls the former) should be called. Something about calling the model directly from a different datatype _feels_ like an end-run around the concept-datatype, like _it_ ought to be the intermediary to get a preflabel from a valueid, but it's hard to justify that case given that it simply calls the model method.

Reply:
I think it's better just to use the concept method than the concept datatype. If the resource instance datatype needed a really complex method from the concept datatype, then it might be different because it could require repeating a lot of code. However, in this case the Concept model has exactly what you need without writing anything new. Because of that, using the datatype isn't any DRYer than just using the Concept method. Instead, it  adds an extra layer of code between the resource-instance datatype and the Concept method.

When we move over to using controlled lists in v8, things should be pretty similar. The ControlledList model could just have a method that returns the prefLabel. The datatype doesn't actually need a 'get_pref_label' method because the tile actually stores the label in its data. So when we start using controlled list values to describe relationships then could also use the ControlledList model like you do the Concept model.

Reply:
@chiatt changes pushed to latest